sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -3,27 +3,89 @@
 
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include <internal/contract.hpp>
 
+using namespace std;
+
 namespace azure
 {
 namespace core
 {
 namespace http
 {
 
-class http_request
+namespace http_method
 {
+enum HttpMethod
+{
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
```

Comment:
>Header [](start = 6, length = 6)

Are we going to allow two classes in one `.h` or `.hpp` file?

Reply:
Let's not.

Reply:
I would say yes, but let's review it in next meeting together.
I would let all related classes to be declared within one same header.
Then we can have multiple .cpp files to define each class.
With this, customer only needs to add one header file (like http.hpp) to get all the http functionality. Instead of learning the list of headers he needs to add (request, header, etc etc)

Reply:
We should minimize the number of headers where it makes sense.  A large class gets its own header file.  A small class used privately within a class can be defined in the header.

Note we can still have one public header "http.hpp" which the customer includes.  The "http.hpp" header then has all the sub-headers included.  The customer gets a simple solution and we get maintainable header files.

Reply:
> With this, customer only needs to add one header file (like http.hpp) to get all the http functionality. Instead of learning the list of headers he needs to add (request, header, etc etc)

> We should minimize the number of headers where it makes sense.

That's a great point that I didn't think about. I agree that having a single class per cpp file makes sense. Having "groups" of related types together in a single header file is a good idea to make it easier for customers to include.